File: app/src/main/java/com/spidroid/starry/ui/search/RecentSearchAdapter.java
package com.spidroid.starry.ui.search;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.spidroid.starry.R;
import java.util.List;

public class RecentSearchAdapter extends RecyclerView.Adapter<RecentSearchAdapter.ViewHolder> {

    private List<String> searchTerms;
    private final OnHistoryInteractionListener listener;

    public interface OnHistoryInteractionListener {
        void onTermClicked(String term);
        void onRemoveClicked(String term);
    }

    public RecentSearchAdapter(List<String> terms, OnHistoryInteractionListener listener) {
        this.searchTerms = terms;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recent_search, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        String term = searchTerms.get(position);
        holder.searchTerm.setText(term);
        holder.itemView.setOnClickListener(v -> listener.onTermClicked(term));
        holder.removeButton.setOnClickListener(v -> listener.onRemoveClicked(term));
    }

    @Override
    public int getItemCount() {
        return searchTerms.size();
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        TextView searchTerm;
        ImageButton removeButton;
        ViewHolder(@NonNull View itemView) {
            super(itemView);
            searchTerm = itemView.findViewById(R.id.tv_search_term);
            removeButton = itemView.findViewById(R.id.btn_remove_term);
        }
    }
}

File: app/src/main/java/com/spidroid/starry/ui/search/RecentSearchAdapter.kt
package com.spidroid.starry.ui.search

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.spidroid.starry.R

class RecentSearchAdapter(
    private val searchTerms: MutableList<String?>,
    private val listener: OnHistoryInteractionListener
) : RecyclerView.Adapter<RecentSearchAdapter.ViewHolder?>() {
    interface OnHistoryInteractionListener {
        fun onTermClicked(term: String?)
        fun onRemoveClicked(term: String?)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.getContext())
            .inflate(R.layout.item_recent_search, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val term = searchTerms.get(position)
        holder.searchTerm.setText(term)
        holder.itemView.setOnClickListener(View.OnClickListener { v: View? ->
            listener.onTermClicked(
                term
            )
        })
        holder.removeButton.setOnClickListener(View.OnClickListener { v: View? ->
            listener.onRemoveClicked(
                term
            )
        })
    }

    override fun getItemCount(): Int {
        return searchTerms.size
    }

    internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var searchTerm: TextView
        var removeButton: ImageButton

        init {
            searchTerm = itemView.findViewById<TextView>(R.id.tv_search_term)
            removeButton = itemView.findViewById<ImageButton>(R.id.btn_remove_term)
        }
    }
}